src/map/script.cpp

```diff
@@ -16855,7 +16855,7 @@ BUILDIN_FUNC(setnpcdisplay)
 	}
 
 	// update npc
-	if( newname )
+	if( newname && nd->bl.m != -1 )
```

Comment:
Please do not do this here. Do it inside the npc_setdisplayname function please, just in case it is called from somewhere else in the future.